sdk/core/System.ClientModel/src/Convenience/ClientResult.cs

```diff
@@ -11,18 +11,16 @@ namespace System.ClientModel;
 /// </summary>
 public class ClientResult
 {
-    private readonly PipelineResponse _response;
+    private PipelineResponse? _response;
 
     /// <summary>
     /// Create a new instance of <see cref="ClientResult"/> from a service
     /// response.
     /// </summary>
     /// <param name="response">The <see cref="PipelineResponse"/> received
     /// from the service.</param>
-    protected ClientResult(PipelineResponse response)
+    protected ClientResult(PipelineResponse? response)
```

Comment:
We would only include this change if we have good reason to believe we'll want it for pageable/operation types in the future.